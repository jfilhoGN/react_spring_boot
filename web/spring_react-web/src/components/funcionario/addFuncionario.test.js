import React from 'react';
import {shallow} from 'enzyme';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import AddFuncionario from './addFuncionario'

describe('AddFuncionario', () => {

    test('render component', () => {
        const wrapper = shallow(<AddFuncionario match={{params: {id: 1}}}/>);
        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();
    });
    
    test('values of state', () => {
        const wrapper = shallow(<AddFuncionario  match={{params: {id: 10444}}}/>)
        expect(wrapper.state()).toMatchSnapshot();
    });

    test('input text nome', () => {
        const wrapper = shallow(<AddFuncionario match={{params:{id:1}}}/>);
        wrapper.find('#nome').simulate('change', { target: { value: 'John'} });
        expect(wrapper.state().nome).toEqual('John');
    });

    test('input text email', () => {
        const wrapper = shallow(<AddFuncionario match={{params:{id:1}}}/>);
        wrapper.find('#email').simulate('change', { target: { value: 'john@123.com'} });
        expect(wrapper.state().email).toEqual('john@123.com');
    });

    test('input text cpf', () => {
        const wrapper = shallow(<AddFuncionario match={{params:{id:1}}}/>);
        wrapper.find('#cpf').simulate('change', { target: { value: '000'} });
        expect(wrapper.state().cpf).toEqual('000');
    });

    test('input text senha', () => {
        const wrapper = shallow(<AddFuncionario match={{params:{id:1}}}/>);
        wrapper.find('#senha').simulate('change', { target: { value: '123mudar'} });
        expect(wrapper.state().senha).toEqual('123mudar');
    });

    test('button addFuncionario', () => {
        const onButtonClick = sinon.spy();
        const wrapper = shallow(<AddFuncionario match={{params:{id:1}}} />);
        wrapper.find('.buttonAdd').simulate('click', { preventDefault() {} });
        expect(onButtonClick.notCalled).toBe(true);
    });

    test('input text editNome', () => {
        const wrapper = shallow(<AddFuncionario match={{params:{id:1}}}/>);
        wrapper.setState({hasEdit:true});
        wrapper.find('#nomeEdit').simulate('change', { target: { value: 'John'} });
        expect(wrapper.state().nome).toEqual('John');
    });
    
    test('input text editEmail', () => {
        const wrapper = shallow(<AddFuncionario match={{params:{id:1}}}/>);
        wrapper.setState({hasEdit:true});
        wrapper.find('#emailEdit').simulate('change', { target: { value: 'john@123.com'} });
        expect(wrapper.state().email).toEqual('john@123.com');
    });

    test('input text editCpf', () => {
        const wrapper = shallow(<AddFuncionario match={{params:{id:1}}}/>);
        wrapper.setState({hasEdit:true});
        wrapper.find('#cpfEdit').simulate('change', { target: { value: '999'} });
        expect(wrapper.state().cpf).toEqual('999');
    });

    test('button updateFuncionario', () => {
        const onButtonClick = sinon.spy();
        const wrapper = shallow(<AddFuncionario match={{params:{id:1}}} />);
        wrapper.setState({hasEdit:true});
        wrapper.find('.buttonUpdate').simulate('click', { preventDefault() {} });
        expect(onButtonClick.notCalled).toBe(true);
    });

    test('render dialog', () => {
        const wrapper = shallow(<AddFuncionario match={{params:{id:1}}}/>);
        wrapper.setState({hasFuncionario:true});
        expect(wrapper).toMatchSnapshot();
    })

    test('button sim Dialog', () => {
        const onButtonClick = sinon.spy();
        const wrapper = shallow(<AddFuncionario match={{params:{id:40}}}/>);
        
    });

    //handleUpdate
    //componentDidMount
    
});