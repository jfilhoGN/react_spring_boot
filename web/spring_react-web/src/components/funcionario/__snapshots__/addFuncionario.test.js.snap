// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFuncionario render component 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <div>
      <Row>
        <form
          className="col s12"
          onSubmit={[Function]}
        >
          <Input
            checked={false}
            id="nome"
            label="Nome"
            name="nome"
            onChange={[Function]}
            placeholder="John Doe"
            s={12}
            type="text"
          >
            <Icon
              small={true}
            >
              person
            </Icon>
          </Input>
          <Input
            checked={false}
            id="email"
            label="Email"
            name="email"
            onChange={[Function]}
            placeholder="johndoe@john.com"
            s={12}
            type="text"
          >
            <Icon
              small={true}
            >
              email
            </Icon>
          </Input>
          <Input
            checked={false}
            id="cpf"
            label="CPF"
            name="cpf"
            onChange={[Function]}
            placeholder="111.111.111-11"
            s={12}
            type="text"
          >
            <Icon
              small={true}
            >
              assignment_ind
            </Icon>
          </Input>
          <Input
            checked={false}
            id="senha"
            label="Senha"
            name="senha"
            onChange={[Function]}
            placeholder="123mudar"
            s={12}
            type="password"
          >
            <Icon
              small={true}
            >
              lock
            </Icon>
          </Input>
          <Col
            m={12}
            s={12}
          >
            <Button
              className="buttonAdd grey darken-3 btn-small"
              name="action"
              node="button"
              onClick={[Function]}
              type="submit"
            >
              <i
                className="material-icons"
              >
                add
              </i>
            </Button>
          </Col>
        </form>
      </Row>
    </div>
  </div>
  <div
    className="card-reveal"
  >
    <span
      className="card-title grey-text text-darken-4"
    >
      <Icon
        right={true}
      >
        close
      </Icon>
    </span>
  </div>
</div>
`;

exports[`AddFuncionario render dialog 1`] = `
<div>
  <WithStyles(Dialog)
    aria-describedby="alert-dialog-description"
    aria-labelledby="alert-dialog-title"
  >
    <WithStyles(DialogTitle)
      id="alert-dialog-title"
    >
      Funcionário Cadastrado
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <WithStyles(DialogContentText)
        id="alert-dialog-description"
      >
        Deseja cadastrar novo funcionário?
      </WithStyles(DialogContentText)>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <Button
        color="primary"
        node="button"
        onClick={[Function]}
      >
        Sim
      </Button>
      <Button
        autoFocus={true}
        color="primary"
        node="button"
        onClick={[Function]}
      >
        Não
      </Button>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;

exports[`AddFuncionario values of state 1`] = `
Object {
  "cpf": "",
  "email": "",
  "hasEdit": false,
  "hasFuncionario": false,
  "id": "",
  "nome": "",
  "senha": "",
}
`;
